source/core/shape.cc

```diff
@@ -522,9 +672,16 @@ Superquadric<dim>::gradient(const Point<dim> &evaluation_point,
       Point<dim> closest_point{};
       this->closest_surface_point(evaluation_point, closest_point);
 
-      Tensor<1, dim> gradient = (evaluation_point - closest_point) /
-                                ((evaluation_point - closest_point).norm() +
-                                 1e-16 * this->effective_radius);
+      Tensor<1, dim> gradient;
+      Point<dim>     centered_point = this->align_and_center(evaluation_point);
+      if (superquadric(centered_point) > 0)
+        gradient = (evaluation_point - closest_point) /
+                   ((evaluation_point - closest_point).norm() +
+                    1e-16 * this->effective_radius);
+      else
+        gradient = -(evaluation_point - closest_point) /
+                   ((evaluation_point - closest_point).norm() +
+                    1e-16 * this->effective_radius);
```

Comment:
```suggestion
      Tensor<1, dim> gradient;
      Point<dim>     centered_point = this->align_and_center(evaluation_point);
      tensor<1,dim> point_distance = evaluation_point - closest_point;
      if (superquadric(centered_point) > 0)
        gradient =  point_distance / (point_distance.norm() +
                    1e-16 * this->effective_radius);
      else
        gradient = - point_distance / (point_distance.norm() +
                    1e-16 * this->effective_radius);
```

Reply:
Is there a reason why you add 1e-16 * this->effective_radius ? If it is for avoiding division by zero, why multiplying by this->effective_radius ? If this is the case, you should replace by DML_MIN

Maybe there's something I don't know...